WordPress/src/main/java/org/wordpress/android/ui/reader/discover/interests/ReaderInterestsFragment.kt

```diff
@@ -24,33 +28,78 @@ class ReaderInterestsFragment : Fragment(R.layout.reader_interests_fragment_layo
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+        initDoneButton()
         initViewModel()
     }
 
+    private fun initDoneButton() {
+        done_button.setOnClickListener {
+            viewModel.onDoneButtonClick()
+        }
+    }
+
     private fun initViewModel() {
-        viewModel = ViewModelProviders.of(this, viewModelFactory).get(ReaderInterestsViewModel::class.java)
+        viewModel = ViewModelProviders.of(this, viewModelFactory)
+            .get(ReaderInterestsViewModel::class.java)
         startObserving()
     }
 
     private fun startObserving() {
         viewModel.uiState.observe(viewLifecycleOwner, Observer { uiState ->
             updateInterests(uiState.interestsUiState)
+            updateDoneButton(uiState.doneButtonUiState)
+        })
+
+        viewModel.navigateToDiscover.observe(viewLifecycleOwner, Observer { event ->
+            event?.getContentIfNotHandled()?.let {
+                navigateToDiscover()
+            }
         })
 
         viewModel.start()
     }
 
+    private fun updateDoneButton(doneButtonUiState: DoneButtonUiState) {
+        with(done_button) {
+            isEnabled = doneButtonUiState.enabled
+            text = getString(doneButtonUiState.titleRes)
+        }
+        uiHelpers.updateVisibility(done_button, doneButtonUiState.visible)
+    }
+
     private fun updateInterests(interestsUiState: List<InterestUiState>) {
         interestsUiState.forEachIndexed { index, interestTagUiState ->
-            val chip = layoutInflater.inflate(R.layout.reader_interest_filter_chip, interests_chip_group, false) as Chip
+            var chip = interests_chip_group.findViewWithTag<Chip>(interestTagUiState.title)
+            if (chip == null) {
+                chip = layoutInflater.inflate(
+                    R.layout.reader_interest_filter_chip,
+                    interests_chip_group,
+                    false
+                ) as Chip
+                with(chip) {
+                    layoutDirection = View.LAYOUT_DIRECTION_LOCALE
+                    tag = interestTagUiState.title
+                    setOnCheckedChangeListener { compoundButton, isChecked ->
+                        if (compoundButton.isPressed) {
+                            viewModel.onInterestAtIndexToggled(index, isChecked)
+                        }
+                    }
+                    interests_chip_group.addView(chip)
```

Comment:
I'm not 100% sure, but could we simplify this by removing this line and setting `attachToRoot` parameter on `layoutInflater.inflate` to `true`?

Reply:
Tried setting true, inflator returns `ChipGroup` instead of `Chip` layout. Can we keep it this way?